### PR TITLE
Wind-Up Boxes

### PR DESCRIPTION
**Pull request by GreatBigFailure**
Introduces two new weapons, both are Jack-in-the-Box style wind-up boxes. One is a makeshift weapon and the other is a traitor uplink item for clowns. I haven't actually added it to the uplink yet as I'd like feedback for balance before I decide the cost and what not.

**Boot-in-a-Box**
It's a spring-loaded boot in a box.

bootinabox_sprites

It's made with a spring (made with metal), a shoe, cardboard, rods, an adhesive (tape, bandage, gum, etc) and a crayon. You need the crayon to write "B" which stands for boot and is important. 

It's probably one of the stronger makeshift weapons but also riskier. You wind it up to 16 times, each wind increases damage slightly, it can't fire below 4, and at higher wind-up it has a tiny knockdown. You can however wind it past max if you keep trying. These wind-ups take much longer and have an exponentially increasing chance of just making it explode. Winding it past maximum increases damage significantly, range, and adds a knockback that throws the target away from the hit.  
I'm going to work on making it able to trigger kicks on machines, like getting free items from vending machines, while this PR is up. I'm not optimistic about it though which is why I'm putting it up now. 

**Punchline**
A clown traitor upgrade of Boot-in-a-Box. 

Punchline_sprites

I don't like items like the bananium sword. Clown traitors should be using the opportunity to annoy people hard enough it requires an antag-pass. Clowns should only kill when it's really funny. Punchline actively makes it harder for you to kill someone but it'll annoy everyone, even the clown. 

It functions similarly to the Boot-in-a-Box. You wind it up to 16, you can wind past that, etc. It also does less damage than the Bootbox. Clowns get a 50% chance to wind twice per use, speeding up reload significantly. The major differences:

- **Instead of throwing the target, Punchline phases them through walls. Like if you threw a wizard mid-jaunt.** 

Like the Bootbox this scales with over-wind. At 16 winds it phases them 2 tiles, enough for a single wall. Each over-wind adds 1 tile as well as projectile range, damage, and a longer knockdown.
  
- **Punchline doesn't explode on failure, it rapidly teleports the clown proportional to how much it was wound up.** 

Each over-wind increases malfunction chance by 10%. For each over-wind it will teleport you once, the teleport's accuracy lowers with windups too. It also stuns you. Malfunctions reset all winds/over-winds.
This means that it's dangerous for clowns to try to space people with Punchline as they'll probably end up in space themselves. They can however wind in a safe location then bring it to a space-able area. Transporting a highly wound Punchline also gives you an emergency teleport since you can only rely on it happening at high over-wind. 

Basically it's very good at helping clowns survive sec chases and annoying the crew. It will however punish the clown for trying to kill people or at least throw them several walls away from the clown to safety. It's also very noisy. 

Collabs, please go over this with a fine tooth comb. Normally I copy paste with slight alterations but this time I wrote more than I copied and I'm sure it shows. There's a lot, a LOT of "if"s in there. I don't want to break the video game, again. 

:cl:
 * rscadd: Added Boot-in-a-Box makeshift weapon. Wind it up and kick people across the room!
* rscadd: Added "Punchline" traitor item. Like Boot-in-a-Box but far, far more honky

### CODE REVIEW

**Comment by GreatBigFailure** [comment]
Just to save you the math for damage balance, for each crank past maximum:
**Boot-in-a-Box**
Failure chance %: 17, 36, 57, 80, 100
Damage: 22, 28, 34, 40, 46

**Punchline**
Failure chance %: 10, 20, etc
Damage: 20, 24, 28, 32, etc capping at 56 to 66 depending on how many double winds clumsy gave them.

**Comment by FudgePucker** [comment]
Punchline is amazing.
10/10

**Comment by Kurfursten** [comment]
Will the boot in a box allow you to avoid damage when kicking machines? If so that could be a concern since a lot of the kick act features are balanced against small amounts of damage

**Comment by GreatBigFailure** [comment]
> Will the boot in a box allow you to avoid damage when kicking machines? If so that could be a concern since a lot of the kick act features are balanced against small amounts of damage

That was the intention. The original idea was a box you wind-up, place down, and are able to activate with a signaler for long range kicking. That wouldn't work with how this version turned out, so if I still like the idea I could make another type that's much more expensive instead.

**Comment by GreatBigFailure** [comment]
Added it to the uplink. All jobs allowed except Mime because there's too many clown-only items and that's no fun. Costs 14 TC, clown discount is 10. 
14 limits most non-clown combinations except a few intentional ones.
10 lets clowns buy two which allows them to keep one highly wound Punchline for emergencies or have a backup since you drop it after teleporting around. It also lets clowns glue it to their hands and invisible spray it because I'm a monster.

Tell me if you think it's too much/too little.

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -952,6 +952,15 @@ var/list/uplink_items = list()
...
 	jobs_exclusive = list("Mime")
 
+/datum/uplink_item/jobspecific/clown_mime/punchline
+	name = "Punchline"
+	desc = "A high risk high reward abomination combining experimental phazon and bananium technologies. Wind-up Punchline to charge it. Enough charge and your targets will slip through reality. Warning: Forcing wind-ups beyond the limiter may reverse the prototype phazite honkpacitors and disrupt reality around the user."
+	item = /obj/item/weapon/gun/hookshot/whip/clownbox
+	cost = 14
+	discounted_cost = 10
+	jobs_with_discount = list("Clown")
+	jobs_excluded = list("Mime")
```
Comment (Kurfursten) [inline code/datums/uplink_item.dm:962]: why? no other traitor item uses this currently

File: code/modules/projectiles/guns/hookshot.dm [hunk 2]
```diff
@@ -506,3 +506,140 @@
...
+	maxlength = 4 //whips are weird and this ends up being 2
+	hooktype = /obj/item/projectile/hookshot/whip/bootbox
+	var/windUp = 0 //amount of times cranked
+	var/maxWindUp = 16 //threshold for overwind
+	var/overWind = 0 //warning/delay system
+	var/state = 0 //Icon changes with each crank, stolen from the crank cell charger
+	var/springForce = 0 //makes the big kicks
+
+/obj/item/weapon/gun/hookshot/whip/bootbox/update_icon()
+	icon_state = "bootbox[state ? "-1" : "-0"]"
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: you could have written this as just `"bootbox-[state]"`

File: code/modules/projectiles/guns/hookshot.dm [hunk 3]
```diff
@@ -506,3 +506,140 @@
...
+
+/obj/item/weapon/gun/hookshot/whip/bootbox/attack_self(mob/user)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: needs sanity, make sure they're not incapacitated so that you can't wind the bootbox if you died while buckled into a chair

File: code/modules/projectiles/guns/hookshot.dm [hunk 4]
```diff
@@ -506,3 +506,140 @@
...
+	state = !state
+	update_icon()
+	if(prob(windUp*springForce)) //prob 0 before they start forcing it, perfectly safe
+		explosion(src.loc,-1,0,1)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: remove src

File: code/modules/projectiles/guns/hookshot.dm [hunk 5]
```diff
@@ -506,3 +506,140 @@
...
+	name = "boot-in-a-box"
+	icon = 'icons/obj/wind_up.dmi'
+	icon_state = "bootbox-0"
+	item_state = "bootbox"
+	desc = "A box with a spring-loaded boot inside. There is a crank attached to wind it up."
+	fire_action = "activate"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	clumsy_check = 0 //Just makes sense
+	force = 5
+	maxlength = 4 //whips are weird and this ends up being 2
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: instead, make a define like this
`#define WHIPLENGTH * 2`

then do this as `maxlength = 2 WHIPLENGTH` - that makes it immediately evident what this is without need for a comment

File: code/modules/projectiles/guns/hookshot.dm [hunk 6]
```diff
@@ -506,3 +506,140 @@
...
+		explosion(src.loc,-1,0,1)
+		qdel(src)
+		return
+	if(windUp >= maxWindUp && overWind <3) //give them a chance to stop winding in the safety zone. Also largely to delay on-the-run use.
+		user.delayNextAttack(20)
+		playsound(src,'sound/weapons/smash.ogg', 25, 1)
+		overWind++
+		to_chat(user, "<span class='notice'>The crank will barely move.</span>")
+		return
+	if(overWind >=3) //we are no longer in the safety zone. You get one free one before boom risk
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: because of how you've written this, if they have overWind = 2, it will first pass the first if and increase to 3, then trigger the second if

what I would do is remove overwind < 3 from the first if and instead nest the overwind into the other if

```dm
if(windUp >= maxWindUp) //give them a chance to stop winding in the safety zone. Also largely to delay on-the-run use.
		if(overWind<3)
				user.delayNextAttack(20)
				playsound(src,'sound/weapons/smash.ogg', 25, 1)
				overWind++
				to_chat(user, "<span class='notice'>The crank will barely move.</span>")
		else
				to_chat(user, "<span class='notice'>With great difficulty you get the crank moving.</span>")
				etc
	   return
```

File: code/modules/projectiles/guns/hookshot.dm [hunk 7]
```diff
@@ -506,3 +506,140 @@
...
+		to_chat(user, "<span class='notice'>With great difficulty you get the crank moving.</span>")
+		user.delayNextAttack(30)
+		playsound(src,'sound/items/crank.ogg',100,1)
+		update_icon()
+		windUp++
+		springForce++
+		return
+	if(windUp < maxWindUp)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: there should be no need for this, since it must be true if it didn't return above

File: code/modules/projectiles/guns/hookshot.dm [hunk 8]
```diff
@@ -506,3 +506,140 @@
...
+		user.delayNextAttack(5)
+		playsound(src, 'sound/items/crank.ogg',50,1)
+		update_icon()
+		windUp++
+		return
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: no trailing return

File: code/datums/uplink_item.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/datums/uplink_item.dm:962]: Mimes do not tell jokes.

File: code/modules/projectiles/guns/hookshot.dm [hunk 9]
```diff
@@ -506,3 +506,140 @@
...
+
+obj/item/weapon/gun/hookshot/whip/bootbox/Fire(atom/target, mob/living/user, params, reflex =0, struggle =0) //4 winds minimum
+	maxlength = 4
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: resetting maxlength here why?

File: code/modules/projectiles/guns/hookshot.dm [hunk 10]
```diff
@@ -506,3 +506,140 @@
...
+		springForce++
+		return
+	if(windUp < maxWindUp)
+		user.delayNextAttack(5)
+		playsound(src, 'sound/items/crank.ogg',50,1)
+		update_icon()
+		windUp++
+		return
+
+obj/item/weapon/gun/hookshot/whip/bootbox/Fire(atom/target, mob/living/user, params, reflex =0, struggle =0) //4 winds minimum
```
Comment (ShiftyRail) [inline code/modules/projectiles/guns/hookshot.dm]: pls  put a slash before `obj`

File: code/modules/projectiles/guns/hookshot.dm [hunk 11]
```diff
@@ -506,3 +506,140 @@
...
+	icon_state = "clownbox-0"
+	item_state = "clownbox"
+	desc = "Given rich deposits of bananium and phazon beyond most spacemen's wildest dreams, they chose to make this."
+	fire_action = "slip open"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	clumsy_check = 0
+	force = 5
+	maxlength = 6
+	hooktype = /obj/item/projectile/hookshot/whip/clownbox
+	var/windUp = 0
```
Comment (ShiftyRail) [inline code/modules/projectiles/guns/hookshot.dm]: Why not have clown boxes and boot boxes be children of a more general "spring-in-a-box" item?

File: code/modules/projectiles/guns/hookshot.dm [hunk 12]
```diff
@@ -506,3 +506,140 @@
...
+obj/item/weapon/gun/hookshot/whip/bootbox/Fire(atom/target, mob/living/user, params, reflex =0, struggle =0) //4 winds minimum
+	maxlength = 4
+	if(windUp <4)
+		playsound(src, 'sound/items/metal_impact.ogg',25,1)
+		to_chat(user, "<span class='notice'>the spring isn't tight enough to fire</span>")
+		return
+	return..()
+
+/obj/item/weapon/gun/hookshot/whip/clownbox
+	name = "Punchline"
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: is this a proper noun? do you want it to be treated as an improper noun? compare:
`you wind the punchline`
vs
`you wind Punchline`

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 13]
```diff
@@ -743,6 +743,80 @@
...
+			to_chat(user,"<span  class='notice'>You put some spring in your step.</span>")
+		else
+			to_chat(user,"<span  class='notice'>You stuff the [src] firmly into the [I].</span>")
+		qdel(I)
+		I = null
+		qdel(src)
+
+/obj/item/weapon/windup_assembly/attackby(var/obj/item/K, mob/user as mob)
+	switch(windupstate)
+		if(0)
```
Comment (ShiftyRail) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: Defines would be better.

File: code/modules/projectiles/projectile/hookshot.dm [hunk 14]
```diff
@@ -244,3 +244,99 @@
...
+	nodamage = 0
+	damage = 0
+	sharpness = 0
+	kill_count = 20 //range is defined by maxlength so this prevents animation issues
+	failure_message = ""
+	can_tether = FALSE
+	var/windUp = 0
+	var/springForce = 0
+
+/obj/item/projectile/hookshot/whip/bootbox/OnFired()
```
Comment (ShiftyRail) [inline code/modules/projectiles/projectile/hookshot.dm]: Same comment as above, I see no reason for bootboxes and clownboxes to be so different.

File: code/modules/projectiles/guns/hookshot.dm (hunk 11 above)
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: way too much pasting in this implementation, make it a child of the other or make an abstract type of which both are a child

this applies to everywhere you've written the same thing twice - vars, update icon, winding code, etc.

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 15]
```diff
@@ -743,6 +743,80 @@
...
+		/obj/item/stack/medical/bruise_pack/bandaid,
+		/obj/item/weapon/reagent_containers/food/snacks/caramelburger
+		)
+
+/obj/item/weapon/stock_parts/spring/attackby(var/obj/item/I, mob/user)
+	if(istype(I, /obj/item/clothing/shoes))
+		var/obj/item/clothing/shoes/G = I
+		playsound(src,'sound/effects/spring.ogg', 50,1)
+		var/obj/item/weapon/windup_assembly/W = new /obj/item/weapon/windup_assembly
+		user.before_take_item(G) //I have no idea what this does but I'm copy pasting and no one can stop me
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: this isn't necessary

File: code/game/objects/items/weapons/stock_parts.dm [hunk 16]
```diff
@@ -281,3 +281,12 @@
 	desc = "A large piece of equipment used to open a window into the subspace dimension."
 	origin_tech = Tc_MAGNETS + "=5;" + Tc_MATERIALS + "=5;" + Tc_BLUESPACE + "=3"
 	starting_materials = list(MAT_IRON = 50)
+
+//Other
+
+/obj/item/weapon/stock_parts/spring
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/stock_parts.dm]: why is this a stock part? stock parts are used in machines as upgrades

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 17]
```diff
@@ -743,6 +743,80 @@
...
+		/obj/item/taperoll/police,
+		/obj/item/taperoll/engineering,
+		/obj/item/taperoll/atmos,
+		/obj/item/taperoll/syndie,
+		/obj/item/device/label_roll,
+		/obj/item/weapon/glue,
+		/obj/item/weapon/bonegel,
+		/obj/item/stack/package_wrap,
+		/obj/item/stack/medical/bruise_pack/bandaid,
+		/obj/item/weapon/reagent_containers/food/snacks/caramelburger
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: wat

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 18]
```diff
@@ -743,6 +743,80 @@
...
+		)
+
+/obj/item/weapon/stock_parts/spring/attackby(var/obj/item/I, mob/user)
+	if(istype(I, /obj/item/clothing/shoes))
+		var/obj/item/clothing/shoes/G = I
+		playsound(src,'sound/effects/spring.ogg', 50,1)
+		var/obj/item/weapon/windup_assembly/W = new /obj/item/weapon/windup_assembly
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: make this `= new(get_turf(src))`

that way it acts as a fallback in case the user's hands are somehow full (shouldn't, but it's good to be safe against nullspacing)

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 19]
```diff
@@ -743,6 +743,80 @@
...
+		/obj/item/device/label_roll,
+		/obj/item/weapon/glue,
+		/obj/item/weapon/bonegel,
+		/obj/item/stack/package_wrap,
+		/obj/item/stack/medical/bruise_pack/bandaid,
+		/obj/item/weapon/reagent_containers/food/snacks/caramelburger
+		)
+
+/obj/item/weapon/stock_parts/spring/attackby(var/obj/item/I, mob/user)
+	if(istype(I, /obj/item/clothing/shoes))
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: include 
`if(user.drop_item(I))` at some point, this checks to make sure you can set down the shoes, that way you can't get rid of glued shoes by putting them into a bootbox

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 20]
```diff
@@ -743,6 +743,80 @@
...
+		qdel(I)
+		I = null
+		qdel(src)
+
+/obj/item/weapon/windup_assembly/attackby(var/obj/item/K, mob/user as mob)
+	switch(windupstate)
+		if(0)
+			if(istype(K, /obj/item/stack/sheet/cardboard))
+				var/obj/item/stack/sheet/cardboard/C = K
+				icon_state = "windup_assembly_1"
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: just make the icon state `windup_assembly_[windupstate+1]` (or rename the states)

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 21]
```diff
@@ -743,6 +743,80 @@
...
+				windupstate = 1
+				playsound(src,'sound/items/can_crushed.ogg', 50,1)
+				to_chat(user,"<span  class='notice'>You build a box around the assembly.</span>")
+				C.use(5)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: you don't check the number, so you can do this with any amount of cardboard! `if(C.use(5))` will work

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 22]
```diff
@@ -743,6 +743,80 @@
...
+		if(1)
+			if(istype(K, /obj/item/stack/rods))
+				var/obj/item/stack/rods/C = K
+				icon_state = "windup_assembly_2"
+				windupstate = 2
+				playsound(src,'sound/items/crank.ogg', 50,1)
+				to_chat(user,"<span  class='notice'>You add a crank to the box.</span>")
+				C.use(3)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: same as above! you can do this with any number of rods

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 23]
```diff
@@ -743,6 +743,80 @@
...
+		if(2)
+			if(is_type_in_list(K, windup_adhesive))
+				icon_state = "windup_assembly_3"
+				windupstate = 3
+				playsound(src,'sound/items/poster_ripped.ogg', 50,1)
+				to_chat(user,"<span  class='notice'>You patch up and reinforce the box.</span>")
+				user.drop_item(K, force_drop = 1)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: you don't check drop item, so they can use it even if it's glued to them

File: code/datums/uplink_item.dm (hunk 1 above)
Comment (GreatBigFailure) [reply code/datums/uplink_item.dm:962]: There's a lot of clown/mime only items so I didn't want to limit it to just clown. It's very un-mime though. It screams bike horns all over the place when it malfunctions, etc. It would be weird to be used by a mime.

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 24]
```diff
@@ -743,6 +743,80 @@
...
+				qdel(K)
+		if(3)
+			if(istype(K, /obj/item/toy/crayon))
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: same, not checking if you can drop it

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 25]
```diff
@@ -743,6 +743,80 @@
...
+				playsound(src,'sound/misc/balloon_twist_short.ogg', 25,1)
+				to_chat(user,"<span  class='notice'>You write B for boot.</span>")
+				new /obj/item/weapon/gun/hookshot/whip/bootbox(get_turf(src.loc))
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: put in hands?

File: code/modules/projectiles/guns/hookshot.dm (hunk 5 above)
Comment (GreatBigFailure) [reply code/modules/projectiles/guns/hookshot.dm]: That was more for people judging balance/myself to remember. It has 4 range like the bullwhip has 5 but it counts the tile it was fired on and one right after the projectile ends. So the bullwhip travels 3. That doesn't actually do anything.

File: code/modules/projectiles/projectile/hookshot.dm [hunk 26]
```diff
@@ -244,3 +244,99 @@
...
+	..()
+	var/obj/item/weapon/gun/hookshot/whip/bootbox/T = shot_from
+	if(istype(T))
+		src.windUp = T.windUp
```
Comment (Kurfursten) [inline code/modules/projectiles/projectile/hookshot.dm]: remove src

File: code/modules/projectiles/projectile/hookshot.dm [hunk 27]
```diff
@@ -244,3 +244,99 @@
...
+		src.springForce = T.springForce //inherits all the oomph from the box itself
+		damage = (windUp + springForce*5)
+		T.maxlength += springForce
```
Comment (Kurfursten) [inline code/modules/projectiles/projectile/hookshot.dm:269]: well this does explain you resetting maxlength above, but it does mean that you can't really varedit these objects to go farther, this doesn't seem like the best implmenetation

File: code/modules/projectiles/projectile/hookshot.dm [hunk 28]
```diff
@@ -244,3 +244,99 @@
...
+				endLocation = get_ranged_target_turf(Q, throwdir, springForce)
+				K.throw_at(endLocation,springForce,windUp)
+				K.Knockdown(2+springForce) //Gotta be really tempting to blow your hands up
+				if (prob(15*springForce))
+					explosion(K.loc,-1,0,0)
+					explosion(T.loc,-1,0,1)
+					qdel(T)
+
+
+/obj/item/projectile/hookshot/whip/clownbox
```
Comment (Kurfursten) [inline code/modules/projectiles/projectile/hookshot.dm]: same as above, you can likely inherit most of this

File: code/modules/projectiles/guns/hookshot.dm (hunk 9 above)
Comment (GreatBigFailure) [reply code/modules/projectiles/guns/hookshot.dm]: SpringForce increases maxLength so if I didn't reset it at some point the thing would just keep getting longer range. It was just the most efficient place to put it since if I had it on the windup proc it would need to trigger a bunch. This way it only triggers once and then the range is recalculated.

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm (hunk 17 above)
Comment (GreatBigFailure) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: Anything you can use as tape or glue. An adhesive.

File: code/datums/uplink_item.dm (hunk 1 above)
Comment (Kurfursten) [reply code/datums/uplink_item.dm:962]: I don't get why it makes sense for mimes to buy a banana grenade and banana gun but not this

Or why it makes more sense for the chaplain or head of security or whatever to get this

File: code/modules/projectiles/guns/hookshot.dm (hunk 5 above)
Comment (Kurfursten) [reply code/modules/projectiles/guns/hookshot.dm]: then define whiplength as `+2`

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm (hunk 17 above)
Comment (Kurfursten) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: no, I was commenting on using a random burger as adhesive

File: code/datums/uplink_item.dm (hunk 1 above)
Comment (GreatBigFailure) [reply code/datums/uplink_item.dm:962]: The former mostly doesn't make sense. Though at least bananas are quiet. 
The latter also doesn't make sense but it's more fun. Clowns already get a lot of benefit vs anyone else. Not only is it a lot cheaper for them but they reload it much faster. Really I'd say it isn't a worthwhile item for anyone but clowns, I just didn't want to limit it because fun. 
The mime exclusion is specifically because it's very loud. It makes horn sounds when you wind it, party favor sounds when you fire it. If a mime uses it and it malfunctions at high springForce it should warrant a vow break.

File: code/modules/projectiles/projectile/hookshot.dm (hunk 27 above)
Comment (GreatBigFailure) [reply code/modules/projectiles/projectile/hookshot.dm:269]: Do you mean you'd prefer it doesn't increase range with springForce or just that I should make it more flexible?

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm (hunk 17 above)
Comment (GreatBigFailure) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: Caramel is sticky

File: code/modules/projectiles/guns/hookshot.dm (hunk 2 above)
Comment (jknpj) [inline code/modules/projectiles/guns/hookshot.dm]:
```suggestion
	icon_state = "bootbox-[state]"
```

File: code/modules/projectiles/guns/hookshot.dm (hunk 8 above)
Comment (jknpj) [inline code/modules/projectiles/guns/hookshot.dm]:
```suggestion
```

File: code/modules/projectiles/guns/hookshot.dm (hunk 10 above)
Comment (jknpj) [inline code/modules/projectiles/guns/hookshot.dm]:
```suggestion
/obj/item/weapon/gun/hookshot/whip/bootbox/Fire(atom/target, mob/living/user, params, reflex =0, struggle =0) //4 winds minimum
```

File: code/modules/projectiles/guns/hookshot.dm (hunk 12 above)
Comment (jknpj) [inline code/modules/projectiles/guns/hookshot.dm]:
```suggestion
	name = "\improper Punchline"
```

File: code/modules/projectiles/guns/hookshot.dm [hunk 29]
```diff
@@ -506,3 +506,146 @@
...
+	hooktype = /obj/item/projectile/hookshot/whip/windup_box/bootbox
+	lengthDecider = 4
+	fireSound = 'sound/effects/fence_smash.ogg'
+	fireVolume = 50
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox/update_icon()
+	icon_state = "bootbox-[state]"
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox/attack_self(mob/user)
+	..()
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: this needs to be 
```
if(..())
  return
```
then at line 538, add return 1

File: code/modules/projectiles/guns/hookshot.dm [hunk 30]
```diff
@@ -506,3 +506,146 @@
...
+	var/minWindUp = 4
+	var/list/fireSound = null
+	var/fireVolume = 0
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/New()
+	..()
+	maxlength = lengthDecider
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/attack_self(mob/user)
+	if(user.stat != CONSCIOUS)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: incapacitated() is better, it checks things like stunned

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm (hunk 17 above)
Comment (Kurfursten) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: Why not other caramel foods? Why not other sticky foods? Just seems weird and unimmersive, like if you could us spaghetti to "tie it on"

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 31]
```diff
@@ -743,6 +743,87 @@
...
+		if(WINDUP_RODS)
+			if(istype(K, /obj/item/stack/rods))
+				var/obj/item/stack/rods/C = K
+				if(C.use(3))
+					windupState = WINDUP_ADHESIVE
+					playsound(src,'sound/items/crank.ogg', 50,1)
+					to_chat(user,"<span  class='notice'>You add a crank to the box.</span>")
+					icon_state = "windup_assembly_rods"
+		if(WINDUP_ADHESIVE)
+			if(is_type_in_list(K, windup_adhesive) && user.drop_item(K, loc))
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: drop_item to its intended location (src not loc)

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm (hunk 23 above)
Comment (Kurfursten) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: you don't need a second drop_item here

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 32]
```diff
@@ -743,6 +743,87 @@
...
+				windupState = WINDUP_CRAYON
+				playsound(src,'sound/items/poster_ripped.ogg', 50,1)
+				to_chat(user,"<span  class='notice'>You patch up and reinforce the box.</span>")
+				user.drop_item(K, force_drop = 1)
+				qdel(K)
+				icon_state = "windup_assembly_adhesive"
+		if(WINDUP_CRAYON)
+			if(istype(K, /obj/item/toy/crayon) && user.drop_item(K,loc))
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: same as above

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 33]
```diff
@@ -743,6 +743,87 @@
...
+		/obj/item/stack/medical/bruise_pack/bandaid,
+		/obj/item/weapon/reagent_containers/food/snacks/caramelburger
+		)
+
+/obj/item/spring/attackby(var/obj/item/I, mob/user)
+	if((istype(I, /obj/item/clothing/shoes)) && (user.drop_item(I)))
+		var/obj/item/clothing/shoes/G = I
+		playsound(src,'sound/effects/spring.ogg', 50,1)
+		var/obj/item/weapon/windup_assembly/W = new /obj/item/weapon/windup_assembly(get_turf(src))
+		user.before_take_item(G)
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: not necessary

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm [hunk 34]
```diff
@@ -743,6 +743,87 @@
...
+		user.before_take_item(src)
+		user.put_in_hands(W)
+		if(prob(10))
+			to_chat(user,"<span  class='notice'>You put some spring in your step.</span>")
+		else
+			to_chat(user,"<span  class='notice'>You stuff the [src] firmly into the [I].</span>")
+		qdel(I)
+		I = null
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: not necessary

File: code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm (hunk 17 above)
Comment (GreatBigFailure) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: I couldn't think of any other significantly sticky foods. I just added it as a goofy little secret. Like 3 years from now some dude gets a smile because he finds out you can make them with caramel burgers.
I did actually consider spaghetti but it seemed like a bit much. Can't we cuff people with spaghetti? I should add spaghetti

Comment (Kurfursten) [reply code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm]: You can but it has a resist time of 1 second to represent that it's a bad idea, since this doesn't fall apart based on the components you use, it seems pretty questionable

File: code/modules/projectiles/guns/hookshot.dm [hunk 35]
```diff
@@ -506,3 +506,148 @@
...
+	lengthDecider = 4
+	fireSound = 'sound/effects/fence_smash.ogg'
+	fireVolume = 50
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox/update_icon()
+	icon_state = "bootbox-[state]"
+
+/obj/item/weapon/gun/hookshot/whip/windup_box/bootbox/attack_self(mob/user)
+	if(..())
+		return
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/hookshot.dm]: this technically works for now but it would be good if you made it return 1 in case there is ever a child of bootbox (for the same reasons
